### PR TITLE
Add hashicorp vault

### PR DESCRIPTION
**Pull request by gavin-stackrox**

### CODE REVIEW

File: images/stackrox-test.Dockerfile [hunk 1]
```diff
@@ -142,6 +142,18 @@ RUN set -ex \
   && rm -rf "shellcheck-v${SHELLCHECK_VERSION}" \
   && shellcheck --version
 
+# Install hashicorp vault
+ARG VAULT_VERSION=1.12.1
+ARG VAULT_SHA256=839fa81eacd250e0b0298e518751a792cd5d7194650af78cf5da74d7b7b1e5fb
+RUN set -ex \
+  && wget --quiet "https://releases.hashicorp.com/vault/1.12.1/vault_${VAULT_VERSION}_linux_amd64.zip" \
```
Comment (porridge) [inline images/stackrox-test.Dockerfile:149]: Do we have `curl` at this stage? I think it has some nice `--retry` flags that could help reduce possible flakes.
Also:
```suggestion
  && wget --quiet "https://releases.hashicorp.com/vault/${VAULT_VERSION}/vault_${VAULT_VERSION}_linux_amd64.zip" \
```

File: images/stackrox-test.Dockerfile [hunk 2]
```diff
@@ -142,6 +142,18 @@ RUN set -ex \
...
+  && sha256sum --check --status <<< "${VAULT_SHA256}  vault_${VAULT_VERSION}_linux_amd64.zip" \
+  && unzip "vault_${VAULT_VERSION}_linux_amd64.zip" \
+  && strip "vault" \
+  && cp "vault" /usr/bin/vault \
```
Comment (porridge) [inline images/stackrox-test.Dockerfile:153]: No need to `chmod +x`?

File: images/stackrox-test.Dockerfile [hunk 3]
```diff
@@ -142,6 +142,18 @@ RUN set -ex \
...
+  && rm "vault_${VAULT_VERSION}_linux_amd64.zip" \
```
Comment (porridge) [inline images/stackrox-test.Dockerfile:154]: `rm` the `vault` in $CWD too, or change the `cp` above to `mv`?

**Review by porridge — APPROVED** [review]

File: images/stackrox-test.Dockerfile (hunk 3 above)
Comment (gavin-stackrox) [reply images/stackrox-test.Dockerfile:154]: oops. again. thanks

**Pull request merged**